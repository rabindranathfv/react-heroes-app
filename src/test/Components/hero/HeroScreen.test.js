import { mount } from 'enzyme';
import { MemoryRouter, Routes, Route } from 'react-router-dom';

import { AuthContext } from './../../../auth/authContext';
import { HeroScreen } from './../../../Components/hero/HeroScreen';

// !important when you are using jest.mock remenber use mock<FunctionName> for easily mock without break scope variables
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('HeroScreen', () => { 

    const contextVal = {
        user: {
            name: 'rabindranath'
        },
        dispatch: jest.fn()
    }

   it('Should render default view with no hero because does not exist', () => {
    const wrapper = mount(
        <AuthContext.Provider value={contextVal} > 
            <MemoryRouter initialEntries={['/hero/']}>
                <Routes>
                    <Route path='/hero' element={ <HeroScreen />} />
                    <Route path='/' element={ <h1>No Hero id </h1>} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>   
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('No Hero id');
   }); 

   it('Should render HeroScreen with hero', () => {

    const wrapper = mount(
        <AuthContext.Provider value={contextVal} > 
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Routes>
                    <Route path='/hero/:heroeId' element={ <HeroScreen />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>   
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.row').exists()).toBeTruthy();
    expect(wrapper.find('img').prop('alt').trim()).toBe('Spider Man');
    expect(wrapper.find('.list-group-item').at(0).text().trim()).toBe('Alter ego: Peter Parker');
  }); 

  it('Should render HeroScreen and go back screen', () => {

    const wrapper = mount(
        <AuthContext.Provider value={contextVal} > 
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Routes>
                    <Route path='/hero/:heroeId' element={ <HeroScreen />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>   
    );

    wrapper.find('button').simulate('click');

    expect(wrapper).toMatchSnapshot();
    expect(mockNavigate).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
});