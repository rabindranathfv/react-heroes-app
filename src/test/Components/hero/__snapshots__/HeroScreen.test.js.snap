// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroScreen Should render HeroScreen and go back screen 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Spider Man"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="/assets/marvel-spider.jpg"
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
      Spider Man
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
         
        <b>
          Alter ego:
        </b>
         
        Peter Parker
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
          Publisher:
        </b>
         
        Marvel Comics
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
          First Appearance:
        </b>
         
        Amazing Fantasy #15
         
      </li>
    </ul>
    <h5
      className="mt-3"
    >
      Characters
    </h5>
    <p>
      Peter Parker
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Regresar
    </button>
  </div>
</div>
`;

exports[`HeroScreen Should render HeroScreen with hero 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Spider Man"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="/assets/marvel-spider.jpg"
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
      Spider Man
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
         
        <b>
          Alter ego:
        </b>
         
        Peter Parker
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
          Publisher:
        </b>
         
        Marvel Comics
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
          First Appearance:
        </b>
         
        Amazing Fantasy #15
         
      </li>
    </ul>
    <h5
      className="mt-3"
    >
      Characters
    </h5>
    <p>
      Peter Parker
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Regresar
    </button>
  </div>
</div>
`;

exports[`HeroScreen Should render default view with no hero because does not exist 1`] = `
<h1>
  No Hero id 
</h1>
`;
